src/components/Modals/FiatRamps/fiatRampProviders/onramper.ts

```diff
@@ -0,0 +1,120 @@
+import { adapters, AssetId } from '@shapeshiftoss/caip'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { uniqBy } from 'lodash'
+import { logger } from 'lib/logger'
+
+import { FiatRampAsset } from '../FiatRampsCommon'
+
+const moduleLogger = logger.child({
+  namespace: ['Modals', 'FiatRamps', 'fiatRampProviders', 'OnRamper'],
+})
+
+// Non-exhaustive required types definition. Full reference: https://github.com/onramper/widget/blob/master/package/src/ApiContext/api/types/gateways.ts
+type OnRamperGatewaysResponse = {
+  gateways: GatewayItem[]
+  icons: TokenIconMap
+}
+
+type TokenIconMap = {
+  [key: string]: IconGatewaysResponse
+}
+
+type Currency = {
+  code: string
+  id: string
+  network?: string
+  displayName?: string
+}
+
+type IconGatewaysResponse = {
+  name: string
+  icon: string
+  symbol?: string
+}
+
+type GatewayItem = {
+  identifier: string
+  cryptoCurrencies: Currency[]
+}
+
+export const getOnRamperAssets = async (): Promise<FiatRampAsset[]> => {
+  const data = await (async () => {
+    try {
+      const baseUrl = getConfig().REACT_APP_ONRAMPER_API_URL
+      const apiKey = getConfig().REACT_APP_ONRAMPER_API_KEY
+      const { data } = await axios.get<OnRamperGatewaysResponse>(
+        `${baseUrl}gateways?includeIcons=true`,
+        {
+          headers: {
+            Authorization: `Basic ${apiKey}`,
+          },
+        },
+      )
+
+      return data
+    } catch (e) {
+      moduleLogger.error(e, 'Failed to fetch assets')
+    }
+  })()
+
+  if (!data) return []
+
+  const fiatRampAssets = convertOnRamperDataToFiatRampAsset(data)
+  return fiatRampAssets
+}
+
+const convertOnRamperDataToFiatRampAsset = (
+  response: OnRamperGatewaysResponse,
+): FiatRampAsset[] => {
+  const allCoins = response.gateways
+    .flatMap(q => q.cryptoCurrencies)
+    .map(currency => {
+      return toFiatRampAsset(currency, response.icons)
+    })
+    .filter(p => p !== undefined) as FiatRampAsset[]
+
+  const uniqueCoins = uniqBy(allCoins, 'assetId')
+  return uniqueCoins
+}
+
+function toFiatRampAsset(currency: Currency, icons: TokenIconMap): FiatRampAsset | undefined {
+  const assetId = adapters.onRamperTickerToAssetId(currency.code)
+  if (assetId !== undefined) {
```

Comment:
Can we use a falsy check here instead of explicitly checking for `undefined`?

```suggestion
  if (assetId) {
```